### PR TITLE
Custom mar support

### PR DESCRIPTION
**Pull request by AyushSawant18588**
1) Updated run.sh and kubeflow_inference_run.py to support shorter commit id entered by user
2) MPT-7B's new version added in model_config.json
3) Handler changed to 'auto' device map for loading model
4) Custom model support in download.py and kubeflow_inference_run.py
5) Created 'generate_data_model.py' to store dataclass that contains attributes needed for MAR file generation
6) Introduced type-checking for functions that use dataclass as arguments 
7) Added changes to read batch size, max_batch_delay, response_timeout and initial workers from model config and use those values in config.properties
8) Update readme for cleanup.py path

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -113,7 +113,8 @@ bash $WORK_DIR/llm/run.sh -n llama2_7b -d data/summarize -g 1 -e llm-deploy -f '
 set HOST and PORT
 ```
 export INGRESS_HOST=$(kubectl get po -l istio=ingressgateway -n istio-system -o jsonpath='{.items[0].status.hostIP}')
-export INGRESS_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="http2")].port}')
+
+export INGRESS_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="http2")].nodePort}')
```
Comment (gavrishp) [inline README.md:117]: Update cleanup.py path as well

File: llm/utils/hfutils.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
+"""
```
Comment (gavrishp) [inline llm/utils/hfutils.py:1]: hf_utils?

File: llm/utils/generate_data_model.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
+"""
```
Comment (gavrishp) [inline llm/utils/generate_data_model.py:1]: Good work, Thanks!

File: llm/download.py [hunk 4]
```diff
@@ -216,120 +146,134 @@ class with relevant information.
...
+        f'"1.0":{{"defaultVersion":true,"marName":"{gen_model.model_name}.mar","minWorkers":1,'
         f'"maxWorkers":1,"batchSize":1,"maxBatchDelay":500,"responseTimeout":60}}}}}}}}',
     ]
 
     with open(config_file_path, "a", encoding="utf-8") as config_file:
         config_file.writelines(config_info)
 
 
-def check_if_model_files_exist(dl_model):
+def get_model_files_and_mar(gen_model: GenerateDataModel, params):
```
Comment (gavrishp) [inline llm/download.py:158]: set_model_files_xx? instead of get_xx as we are not returning anything

File: llm/download.py [hunk 5]
```diff
@@ -216,120 +146,134 @@ class with relevant information.
...
+                gen_model.mar_utils.handler_path = os.path.join(
+                    os.path.dirname(__file__),
+                    HANDLER,
                 )
-                sys.exit(1)
+            if gen_model.repo_info.repo_version == "":
+                gen_model.repo_info.repo_version = "1.0"
         else:
             print(
                 "## Please check your model name, it should be one of the following : "
```
Comment (gavrishp) [inline llm/download.py:279]: Add a message , if custom model include to `no_download` flag

File: llm/download.py [hunk 6]
```diff
@@ -338,67 +282,72 @@ class with relevant information.
...
-            print("## Model files do not match HuggingFace repository Files")
-            sys.exit(1)
+        check_if_path_exists(gen_model.mar_utils.model_path, "model_path", is_dir=True)
+        if not gen_model.is_custom:
+            if not check_if_model_files_exist(gen_model):
+                # checking if local model files are same the repository files
+                print("## Model files do not match HuggingFace repository Files")
+                sys.exit(1)
+        else:
+            print(
```
Comment (gavrishp) [inline llm/download.py:340]: Should we do a non-empty folder check?

File: llm/model_config.json [hunk 7]
```diff
@@ -69,7 +65,6 @@
         "repo_id": "gpt2",
         "repo_version": "11c5a3d5811f50298f278a704980280950aedb10",
         "handler": "handler.py",
-        "requirements_file": "model_requirements.txt",
```
Comment (gavrishp) [inline llm/model_config.json:72]: we need these right?

Comment (AyushSawant18588) [reply llm/model_config.json:72]: Custom model will also require it and it is common to all models so I made it as global variable in marsgen.py.

**Review by gavrishp — DISMISSED** [review]

**Comment by gavrishp** [comment]
Add the config properties worker and batch size changes

**Review by gavrishp — APPROVED** [review]

**Review by gavrishp — APPROVED** [review]

**Pull request merged**